File: FlexServ Backend/SpringBoot/AdminService/src/main/java/com/flexserv/dto/response/ServiceProviderResponse.java
package com.flexserv.dto.response;

import java.math.BigDecimal;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ServiceProviderResponse {

    private Long id;

    private Long userId;

    private String userName;

    private String userEmail;

    private String userPhone;

    private String companyName;

    private Integer experienceYears;

    private String bio;

    private Boolean isVerified;

    private BigDecimal rating;

    private Boolean companyAvailable;
}

File: FlexServ Backend/DotNet/AdminService/Dtos/ServiceProviderResponse.cs
using System.Collections.Generic;

namespace AdminService.Dtos
{
    public class ServiceProviderResponse
    {
        public long Id { get; set; }
        public long? UserId { get; set; }
        public string? UserName { get; set; }
        public string? UserEmail { get; set; }
        public string? UserPhone { get; set; }
        public string? CompanyName { get; set; }
        public int? ExperienceYears { get; set; }
        public string? Bio { get; set; }
        public bool? IsVerified { get; set; }
        public decimal? Rating { get; set; }
        public bool? CompanyAvailable { get; set; }
        public List<long> ServiceIds { get; set; } = new List<long>();
        public List<string> ServiceNames { get; set; } = new List<string>();

        public ServiceProviderResponse() { }

        public ServiceProviderResponse(long id, long? userId, string? userName, string? userEmail, string? userPhone,
            string? companyName, int? experienceYears, string? bio, bool? isVerified, decimal? rating, bool? companyAvailable,
            List<long>? serviceIds = null, List<string>? serviceNames = null)
        {
            Id = id;
            UserId = userId;
            UserName = userName;
            UserEmail = userEmail;
            UserPhone = userPhone;
            CompanyName = companyName;
            ExperienceYears = experienceYears;
            Bio = bio;
            IsVerified = isVerified;
            Rating = rating;
            CompanyAvailable = companyAvailable;
            ServiceIds = serviceIds ?? new List<long>();
            ServiceNames = serviceNames ?? new List<string>();
        }
    }
}
